pleiades/vocabularies/vocabularies.py

```diff
@@ -0,0 +1,39 @@
+from zope.component import getUtility
+from zope.interface import directlyProvides
+from zope.schema.interfaces import IVocabularyFactory
+from zope.schema.vocabulary import SimpleTerm
+from zope.schema.vocabulary import SimpleVocabulary
+
+from plone.registry.interfaces import IRegistry
+
+from pleiades.vocabularies.interfaces import IPleiadesSettings
+
+
+def registry_vocabulary(field, value_filter=None):
+    path = field.interface.__identifier__ + '.' + field.__name__
+
+    def vocabulary_factory(context):
+        registry = getUtility(IRegistry)
+        values = registry.get(path, [])
+        values.sort()
+        terms = []
+        for value in values:
+            terms.append(SimpleTerm(
+                value=value['id'],
+                token=value['id'].encode('raw_unicode_escape'),
```

Comment:
Hm, I think produces 8-bit output equivalent to latin-1 with escapes for other characters.

    >>> u'\u00e9'.encode('raw_unicode_escape')
    '\xe9'

    >>> print u'\u00e9'.encode('raw_unicode_escape')
    ?

Seems like ``unicode_escape`` does escaped literals though:

    >>> u'\u00e9'.encode('unicode_escape')
    '\\xe9'
    >>> print u'\u00e9'.encode('unicode_escape')
    \xe9

    >>> b2a_qp(u'\u00e9'.encode('utf-8'))
    '=C3=A9'